autogpt_platform/backend/schema.prisma

```diff
@@ -272,11 +385,220 @@ model UserBlockCredit {
 
   @@id(name: "creditTransactionIdentifier", [transactionKey, userId])
 }
+
+// This is the top level accounting table for a user
+model UserAccounting {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  transactions UserBlockCredit[]
+
+  usdBalance Float @default(0.0)
+
+  subscription   UserSubscription? @relation(fields: [subscriptionId], references: [id])
+  subscriptionId String?
+}
+
+// This table tracks a user's subscription to autogpt
+model UserSubscription {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String
+  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isSubscribed          Boolean   @default(false)
+  subscriptionStartDate DateTime?
+
+  isCancelling        Boolean   @default(false)
+  subscriptionEndDate DateTime?
+
+  subscriptionPlanId String?
+  subscriptionPlan   SubsctiptionPlan? @relation(fields: [subscriptionPlanId], references: [id])
+
+  hasInitiatedChargeBackProcess Boolean          @default(false)
+  chargeBackProcessInitiatedAt  DateTime?
+  UserAccounting                UserAccounting[]
+
+  @@index([isSubscribed, isCancelling])
+}
+
+// This table tracks the different subscription plans available to a user
+model SubsctiptionPlan {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name        String
+  description String
+  planIconUrl String?
+
+  creditsPerMonth Int
+  usdPrice        Float
+
+  isDeleted        Boolean            @default(false)
+  UserSubscription UserSubscription[]
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+//////////////   Store TABLES    /////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model Profile {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String?
+  user   User?   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isGroupProfile Boolean    @default(false)
+  groupId        String?
+  group          UserGroup? @relation(fields: [groupId], references: [id])
+
+  username    String @unique
+  description String
+
+  website  String?
+  github   String?
+  linkedin String?
+  facebook String?
+  twitter  String?
+
+  avatarUrl String?
+
+  @@index([username])
+}
+
+model StoreListing {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  isDeleted  Boolean @default(false)
+  // Not needed but makes lookups faster
+  isApproved Boolean @default(false)
+
+  slug        String
```

Comment:
I'm planning on using `username/slug` or `groupname/slug` as the url so slug does not need to be unique